### ISSUE TITLE
Let people select SSH key

### ISSUE THREAD

**Issue by raisty** [ForkIssues/Tracker#250]
I have several SSH keys depends which Git I am using.
Kraken let people choose the SSH key.
Please, let people choose SSH key.

![screen shot 2018-03-20 at 21 29 40](https://user-images.githubusercontent.com/537310/37681158-29957bce-2c86-11e8-9f6d-dbb73c5deaf1.png)

**Comment by JonDum** [comment]
You can create custom "hosts" to use in your `~/.ssh/config` and then configure your repo to push to that host. Really not any of Fork's business to be providing UI for that.

**Comment by DanPristupov** [comment]
While it's not a direct Fork's problem, it could be nice if Fork could help users to set up things more easily.

Fork lets you choose certificate for GitHub and GitLab already, I just need to implement support for BitBucket. For now, please use `~/.ssh/config` as @JonDum mentioned.

**Comment by raisty** [comment]
Thank you, anyway Bitbucket implementation will be highly appreciated.
Any plans for that?

**Comment by pawelkondraciuk** [comment]
I am using multiple accounts for work/personal stuff with different ssh keys and many times I have pushing problems due to permission issues. It would be cool to select ssh key per repo

**Comment by Stanzilla** [comment]
Well it would certainly be nice to have a SSH key section in the per-repository section

**Comment by ImBoop** [comment]
Any update on this? It'd be nice if we could have a per-repository SSH key.

**Comment by DanPristupov** [comment]
@ImBoop add other SSH key in CLI using `ssh-add -K ~/.ssh/id_rsa` command (https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

**Comment by ImBoop** [comment]
@DanPristupov Why provide an option to select an SSH key if we have to use an external tool to add another, despite there being a drop down? (not to mention how this is problematic to do on non-OpenSSH platforms such as Windows)

For implementation I'd suggest using custom git config field such as `fork.sshkey` or similar to save the config to the repo itself that'd point to the key path, which may make it easier to implement.

**Comment by ivasic** [comment]
> You can create custom "hosts" to use in your `~/.ssh/config` and then configure your repo to push to that host. Really not any of Fork's business to be providing UI for that.

@JonDum any chance you can provide an example for that? I configured a custom key for `bitbucket.org` host and it is respected by `git` when running in terminal but Fork seems to completely ignore it. Example:

```
Host bitbucket.org
    IdentityFile ~/.ssh/id_rsa_custom
    User git
```

edit: actually, Fork only seems to ignore this if the host is `bitbucket.org` as soon as I change it to anything else (e.g. `bitbucket.or` and use this URL in Fork - it uses the correct SSH key. Wild guess here, but maybe Fork is overriding settings for known git hosts like BitBucket (cc @DanPristupov)?

**Comment by sarendipitee** [comment]
I do it like this:

```
Host bbgit
    Hostname bitbucket.org
    IdentityFile ~/.ssh/id_rsa_bitbucket
```

And then I setup my origin using that "fake" host: `git clone git@bbgit:some/repo.git`

As long as you can pull/push on the command line Fork should be able to

**Comment by ivasic** [comment]
Yeah, thanks. That's indeed the workaround I'm using as well. But I'm more concerned that it seems that Fork is doing something strange when I use the real bitbucket hostname. IMO this should be fixed, if that's indeed what's happening

**Comment by clounie** [comment]
Seems like that is happening to me as well.

My `~/.gitconfig`:
```
...
[user]
        name = clounie
...
```

My `~/.ssh/config`:
```
Host github.com-clounie
        Hostname github.com
        User git
        IdentityFile ~/.ssh/my_priv_key
```

Works great with terminal, but Fork bombs even after restart.

**Comment by DanPristupov** [comment]
@clounie ensure that no SSH key is selected in the SSH configuration dialog (Fork -> Configure SSH keys). Otherwise Fork overrides the ssh config.

**Comment by clounie** [comment]
Thanks Dan - none are/were.
<img width="98" alt="image" src="https://user-images.githubusercontent.com/11947182/156836181-725547a8-3d77-45ec-bce4-02fd48e59116.png">

**Comment by DanPristupov** [comment]
Did you try to switch git instance to the system one in the Fork preferences?

**Comment by clounie** [comment]
Sorry - missed this. No change, I just use Fork to manage files and CLI to push/pull right now.

My default was Fork's instance. I tried switching to system instance --> quit fork --> re-opened. Same issue

**Comment by clounie** [comment]
Couple thoughts for context @DanPristupov -
- This is an M1 mac.
- I don't have any lines like this in Terminal but it still works. Don't know if that's zsh (I checked `~/.zprofile` and `~/.zshrc`), zprezto, or macOS becoming smarter.
```
eval $(ssh-agent -s)
ssh-add ~/.ssh/my_key
```
- I did try adding those lines to `~/.zprofile` with Fork using system git client - but that didn't help with this issue (restarted Fork before testing ^)
  - Also - I know that those lines did take affect when I ran them. Got the classic response like `Agent pid 40351 \n Identity added: /Users/MyUser/.ssh/my_key (email@domain.com)`

**Comment by homm** [comment]
I rotated my ssh keys including the new name (`id_ed` instead of `id_rsa`). It was extreme inobvious why Fork tries to find the old key, even after I edit `~/.ssh/config` and even `git pull` works. Only this thread helps me (SSH configuration dialog (Fork -> Configure SSH keys)). For me, Fork should rely on `~/.ssh/config` and `Configure SSH keys` should be in the normal settings dialog, not in separate menu.

**Comment by rursache** [comment]
> [@clounie](https://github.com/clounie) ensure that no SSH key is selected in the SSH configuration dialog (Fork -> Configure SSH keys). Otherwise Fork overrides the ssh config.

just had this issue, fixed by unticking all the ssh keys from that menu. confusing but i wont have to do this ever again. thanks!
